WalletWasabi/Services/WalletService.cs

```diff
@@ -537,55 +534,34 @@ public Operation(Script script, Money amount, string label)
-			{
-				throw new ArgumentException($"{nameof(toSend)} cannot contain negative element.");
-			}
+
+			Guard.ThrowIf(toSend.Any(x => x == null), typeof(ArgumentNullException), $"{nameof(toSend)} cannot contain null element.");
+			Guard.ThrowIf(toSend.Any(x => x.Amount < Money.Zero), typeof(ArgumentException), $"{nameof(toSend)} cannot contain negative element.");
 
 			long sum = toSend.Select(x => x.Amount).Sum().Satoshi;
-			if (sum < 0 || sum > Constants.MaximumNumberOfSatoshis)
-			{
-				throw new ArgumentOutOfRangeException($"{nameof(toSend)} sum cannot be smaller than 0 or greater than {Constants.MaximumNumberOfSatoshis}.");
-			}
+			Guard.ThrowIf(sum < 0 || sum > Constants.MaximumNumberOfSatoshis, typeof(ArgumentOutOfRangeException), $"{nameof(toSend)} sum cannot be smaller than 0 or greater than {Constants.MaximumNumberOfSatoshis}.");
 
 			int spendAllCount = toSend.Count(x => x.Amount == Money.Zero);
-			if (spendAllCount > 1)
-			{
-				throw new ArgumentException($"Only one {nameof(toSend)} element can contain Money.Zero. Money.Zero means add the change to the value of this output.");
-			}
-			if (spendAllCount == 1 && customChange != null)
-			{
-				throw new ArgumentException($"{nameof(customChange)} and send all to destination cannot be specified the same time.");
-			}
+			Guard.ThrowIf(spendAllCount > 1, typeof(ArgumentException), $"Only one {nameof(toSend)} element can contain Money.Zero. Money.Zero means add the change to the value of this output.");
+
+			Guard.ThrowIf(spendAllCount == 1 && customChange != null, typeof(ArgumentException), $"{nameof(customChange)} and send all to destination cannot be specified the same time.");
+
 			Guard.InRangeAndNotNull(nameof(feeTarget), feeTarget, 0, 1008); // Allow 0 and 1, and correct later.
 			if (feeTarget < 2) // Correct 0 and 1 to 2.
 			{
 				feeTarget = 2;
 			}
 			if (subtractFeeFromAmountIndex != null) // If not null, make sure not out of range. If null fee is substracted from the change.
 			{
-				if (subtractFeeFromAmountIndex < 0)
-				{
-					throw new ArgumentOutOfRangeException($"{nameof(subtractFeeFromAmountIndex)} cannot be smaller than 0.");
-				}
-				if (subtractFeeFromAmountIndex > toSend.Length - 1)
-				{
-					throw new ArgumentOutOfRangeException($"{nameof(subtractFeeFromAmountIndex)} can be maximum {nameof(toSend)}.Length - 1. {nameof(subtractFeeFromAmountIndex)}: {subtractFeeFromAmountIndex}, {nameof(toSend)}.Length - 1: {toSend.Length - 1}.");
-				}
+				Guard.ThrowIf(subtractFeeFromAmountIndex < 0, typeof(ArgumentOutOfRangeException), $"{nameof(subtractFeeFromAmountIndex)} cannot be smaller than 0.");
+				Guard.ThrowIf(subtractFeeFromAmountIndex > toSend.Length - 1, typeof(ArgumentOutOfRangeException), $"{nameof(subtractFeeFromAmountIndex)} can be maximum {nameof(toSend)}.Length - 1. {nameof(subtractFeeFromAmountIndex)}: {subtractFeeFromAmountIndex}, {nameof(toSend)}.Length - 1: {toSend.Length - 1}.");
```

Comment:
Here we are exchanging problems. Now we have an incredible insane line length. Please find a another approach to make the code more readeable.

Reply:
Sorry, i was unfair here, it is the message length that is too large. Anyway, you could take the opportunity to improve it a bit.